unitTests/controllers/CategoryFacetQueryControlllerTest.ts

```diff
@@ -0,0 +1,72 @@
+import { CategoryFacetQueryController } from '../../src/controllers/CategoryFacetQueryController';
+import * as Mock from '../MockEnvironment';
+import { CategoryFacet, ICategoryFacetOptions } from '../../src/ui/CategoryFacet/CategoryFacet';
+import { IGroupByRequest } from '../../src/rest/GroupByRequest';
+import { AllowedValuesPatternType } from '../../src/rest/AllowedValuesPatternType';
+import { FakeResults } from '../Fake';
+
+export function CategoryFacetQueryControllerTest() {
+  describe('CategoryFacetQueryController', () => {
+    let testCategoryFacet: Mock.IBasicComponentSetup<CategoryFacet>;
+    let categoryFacetController: CategoryFacetQueryController;
+
+    beforeEach(() => {
+      testCategoryFacet = Mock.optionsComponentSetup<CategoryFacet, ICategoryFacetOptions>(CategoryFacet, {
+        field: '@someField',
+        injectionDepth: 12345
+      });
+      categoryFacetController = new CategoryFacetQueryController(testCategoryFacet.cmp);
+    });
+
+    it('should create a valid group by request when searching', () => {
+      categoryFacetController.searchFacetValues('some value', 15);
+
+      expect(testCategoryFacet.env.searchEndpoint.search).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          groupBy: jasmine.arrayContaining([
+            jasmine.objectContaining({
+              allowedValues: ['*some value*'],
+              allowedValuesPatternType: AllowedValuesPatternType.Wildcards,
+              maximumNumberOfValues: 15,
+              field: testCategoryFacet.cmp.options.field,
+              sortCriteria: 'occurrences',
+              injectionDepth: testCategoryFacet.cmp.options.injectionDepth
+            } as IGroupByRequest)
+          ])
+        })
+      );
+    });
+
+    describe('when searching and getting values back from the endpoint', () => {
+      beforeEach(() => {
+        const fakeResults = FakeResults.createFakeResults();
+        const fakeGroupBy = FakeResults.createFakeHierarchicalGroupByResult('@someField', 'value');
+        fakeGroupBy.values[0].numberOfResults = 123456;
+        fakeGroupBy.values[1].numberOfResults = 123456;
+        fakeGroupBy.values[0].value = 'a|much|longer|path';
+        fakeGroupBy.values[1].value = 'a|shorter|path';
+
+        fakeResults.groupByResults = [fakeGroupBy];
+        (testCategoryFacet.env.searchEndpoint.search as jasmine.Spy).and.returnValue(fakeResults);
+      });
+
+      it('should order by the number of results', async done => {
```

Comment:
I don't think this test can check that the sort is meant to prioritise the number of results, since both values have the same number of results (123456). If you remove the `if` condition on line 46 and the `return` on line 49 of  `CategoryFacetQueryController.ts`, I suspect both this test and the next one on line 62 will still pass.

I think it is important to build values with different number of results.

Reply:
I only modifiy the first 2 values in the fake group by, but there are 10 values total being built. The other ones do not have 123456 number of results, but all different number.